compat/win32/fscache.c

```diff
@@ -204,18 +232,17 @@ static struct fsentry *fsentry_create_list(struct fscache *cache, const struct f
 			dir->len, MAX_PATH - 2, core_long_paths)) < 0)
 		return NULL;
 
-	/*
-	 * append optional '\' and wildcard '*'. Note: we need to use '\' as
-	 * Windows doesn't translate '/' to '\' for "\\?\"-prefixed paths.
-	 */
-	if (wlen)
-		pattern[wlen++] = '\\';
-	pattern[wlen++] = '*';
-	pattern[wlen] = 0;
-
-	/* open find handle */
-	h = FindFirstFileExW(pattern, FindExInfoBasic, &fdata, FindExSearchNameMatch,
-		NULL, FIND_FIRST_EX_LARGE_FETCH);
+	/* handle CWD */
+	if (!wlen) {
+		wlen = GetCurrentDirectoryW(ARRAY_SIZE(pattern), pattern);
+		if (!wlen || wlen >= ARRAY_SIZE(pattern)) {
+			errno = wlen ? ENAMETOOLONG : err_win_to_posix(GetLastError());
+			return NULL;
+		}
+	}
+
+	h = CreateFileW(pattern, FILE_LIST_DIRECTORY, FILE_SHARE_READ | FILE_SHARE_WRITE,
```

Comment:
Please share for delete too "FILE_SHARE_DELETE" to avoid conflicting with someone who has the directory opened for delete though they might not be deleting it.